### PR TITLE
Refactor windows workflow to work on standard windows

### PR DESCRIPTION
**Pull request by thiagocrepaldi**
Currently the windows workflows require bash installed, which is not standard o windows - not to say unusual to have a linux shell as a windows dependency.

it also modifies some syntaxes that seems to fail on windows 11 powershell sessions, like the &&

Lastly, some git authentication hack was modified as it should not be needed to perform git submodule update

### CODE REVIEW

File: .github/workflows/release_win.yml [hunk 1]
```diff
@@ -104,33 +103,42 @@ jobs:
...
+        python -m pip uninstall -y protobuf numpy
+        python -m pip install -q -r requirements-release.txt
         python -m pip install -q onnxruntime==1.17.3
         $Env:ORT_MAX_IR_SUPPORTED_VERSION=9
         $Env:ORT_MAX_ML_OPSET_SUPPORTED_VERSION=3
         $Env:ORT_MAX_ONNX_OPSET_SUPPORTED_VERSION=20
         pytest
+
+    - name: Clean Workspace
+      if: always()
```
Comment (justinchuby) [inline .github/workflows/release_win.yml:142]: Why is this required?

**Comment by justinchuby** [comment]
Lgtm thanks! One comment to clarify the cleaning step. Not sure why it is needed yet.

**Review by xadupre — APPROVED** [review]

**Comment by justinchuby** [comment]
I still think we need to understand why the Clean Workspace step is needed. In theory the checkout workflow should have inserted a clean up step already

File: .github/workflows/release_win.yml [hunk 2]
```diff
@@ -104,33 +103,42 @@ jobs:
...
+      run: |
+        Get-ChildItem -Recurse | Remove-Item -Recurse -Force
```
Comment (justinchuby) [inline .github/workflows/release_win.yml:144]:
```suggestion
```

File: .github/workflows/win_no_exception_ci.yml [hunk 3]
```diff
@@ -79,3 +82,8 @@ jobs:
         if($lastexitcode -ne 0) {
           EXIT 1
         }
+
+    - name: Clean Workspace
+      if: always()
+      run: |
+        Get-ChildItem -Recurse | Remove-Item -Recurse -Force
```
Comment (justinchuby) [inline .github/workflows/win_no_exception_ci.yml:89]:
```suggestion
```

**Review by justinchuby — APPROVED** [review]

File: .github/workflows/release_win.yml (hunk 1 above)
Comment (thiagocrepaldi) [reply .github/workflows/release_win.yml]: Because it is not always guaranteed to have a clean workspace in all github instances.
One example is github enterprise :)

Comment (thiagocrepaldi) [reply .github/workflows/release_win.yml]: sorry, somehow i missed the github notifications for this pr

Comment (justinchuby) [reply .github/workflows/release_win.yml]: I see. check if the updated checkout step solves this? If not feel free to add the step back with comments for future readers. Thanks!
